### PR TITLE
fix(Designer): Add connectionReferenceKeyFormat for hybrid triggers for deserialization

### PR DESCRIPTION
**Pull request by bnarayanared**
1. For deserialization of workflow, adding the HybridTrigger case to extract the connectionReferenceKey
2. For adding the SplitOn value, adding the hybridTrigger connectionReferenceKey type to get the AliasPathParsing

### CODE REVIEW

File: libs/designer/src/lib/core/actions/bjsworkflow/connections.ts [hunk 1]
```diff
@@ -440,3 +443,19 @@ export function getLegacyConnectionReferenceKey(operationDefinition: any): strin
   }
   return referenceKey;
 }
+
+function getHybridTriggerConnectionReferenceKey(operationDefinition: LogicAppsV2.HybridTriggerConnectionInfo): string {
+  let connectionName: string;
+  if (typeof operationDefinition.host.connection.name === 'string') {
```
Comment (preetriti1) [inline libs/designer/src/lib/core/actions/bjsworkflow/connections.ts:449]: do you guys have two formats for a hybrid trigger?

File: libs/utils/src/lib/models/logicAppsV2.ts [hunk 2]
```diff
@@ -100,6 +100,14 @@ export interface OpenApiOperationInputs extends RetryableActionInputs {
   parameters?: any;
 }
 
+export interface HybridTriggerOperationAction extends Trigger {
+  inputs: HybridTriggerConnectionInfo;
```
Comment (preetriti1) [inline libs/utils/src/lib/models/logicAppsV2.ts:104]: Why do you call it action and trigger when this is just trigger.. and this does not seem correct you have other properties also inside inputs no? like the schema

File: libs/designer/src/lib/core/utils/outputs.ts [hunk 3]
```diff
@@ -120,7 +120,7 @@ export const getUpdatedManifestForSplitOn = (manifest: OperationManifest, splitO
       throw new AssertionException(AssertionErrorCode.INVALID_SPLITON, invalidSplitOn);
     }
 
-    const isAliasPathParsingEnabled = manifest.properties.connectionReference?.referenceKeyFormat === ConnectionReferenceKeyFormat.OpenApi;
+    const isAliasPathParsingEnabled = (manifest.properties.connectionReference?.referenceKeyFormat === ConnectionReferenceKeyFormat.OpenApi || manifest.properties.connectionReference?.referenceKeyFormat === ConnectionReferenceKeyFormat.HybridTrigger) ? true : false;
```
Comment (preetriti1) [inline libs/designer/src/lib/core/utils/outputs.ts:123]: why the need for ternary operator when both are boolean evaluations??????

**Review by preetriti1 — CHANGES_REQUESTED** [review]

File: libs/designer/src/lib/core/actions/bjsworkflow/connections.ts (hunk 1 above)
Comment (bnarayanared) [reply libs/designer/src/lib/core/actions/bjsworkflow/connections.ts]: Removed the string check; only one connection type

File: libs/utils/src/lib/models/logicAppsV2.ts (hunk 2 above)
Comment (bnarayanared) [reply libs/utils/src/lib/models/logicAppsV2.ts:104]: Added the other missing properties

File: libs/designer/src/lib/core/utils/outputs.ts (hunk 3 above)
Comment (bnarayanared) [reply libs/designer/src/lib/core/utils/outputs.ts]: Removed

**Review by preetriti1 — APPROVED** [review]

**Pull request merged**